src/compiler/crystal/semantic/call.cr

```diff
@@ -1101,6 +1101,8 @@ class Crystal::Call
 
   def bubbling_exception(&)
     yield
+  rescue ex : Crystal::MacroRaiseException
+    ::raise ex
```

Comment:
I think this should also add information about the instantiation context to the error trace like for `CodeError`.

For the following program, I would like to keep `foo(1)` appearing in the trace:
```cr
def foo(x)
  {% raise "foo" %}
end

foo(1)
```

This is the full error output with and without this patch, each is a single frame. I think the ideal solution would be to have both in the error trace.
```console
$ bin/crystal run .test/macro-raise.cr --error-trace
Using compiled compiler at .build/crystal
In .test/macro-raise.cr:2:6

 2 | {% raise "foo" %}
        ^----
Error: foo
$ crystal run .test/macro-raise.cr --error-trace
In .test/macro-raise.cr:5:1

 5 | foo(1)
     ^--
Error: foo
```

Reply:
Yea I agree. Lines up with my other thought in #13261 too.

Let me play around with it and see if I can get everything in the order we want. There's quite a few different scenarios that determine what the order should be so gotta wrap my head around what all of those are and such.

Reply:
Okay, think I figured that all out. Going to update OP with new before/after traces.